File: src/main/java/de/nordgedanken/auto_hotkey/util/AhkBundle.java
package de.nordgedanken.auto_hotkey.util;

import com.intellij.CommonBundle;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.PropertyKey;

import java.util.ResourceBundle;

public class AhkBundle {
	@NonNls
	protected static final String BUNDLE_NAME = "localization.AhkBundle";

	protected static final ResourceBundle BUNDLE = ResourceBundle.getBundle(BUNDLE_NAME);

	private AhkBundle() {

	}

	/**
	 * Retrieve property value from bundle file
	 *
	 * Ex: AhkBundle.msg("runconfig.configtab.scriptpath.label")
	 */
	public static String msg(@PropertyKey(resourceBundle = BUNDLE_NAME) String key, Object... params) {
		return CommonBundle.message(BUNDLE, key, params);
	}
}

File: src/main/kotlin/com/autohotkey/util/AhkBundle.kt
package com.autohotkey.util

import com.intellij.AbstractBundle
import org.jetbrains.annotations.NonNls
import org.jetbrains.annotations.PropertyKey
import java.util.ResourceBundle

object AhkBundle {
    @NonNls
    internal const val BUNDLE_NAME = "localization.AhkBundle"
    private val BUNDLE = ResourceBundle.getBundle(BUNDLE_NAME)

    /**
     * Retrieve property value from bundle file
     *
     * Ex: AhkBundle.msg("runconfig.configtab.scriptpath.label")
     */
    fun msg(@PropertyKey(resourceBundle = BUNDLE_NAME) key: String, vararg params: Any?): String {
        return AbstractBundle.message(BUNDLE, key, *params)
    }
}
